project-base/storefront/hooks/cart/useOrderPagesAccess.ts

```diff
@@ -0,0 +1,36 @@
+import { useCurrentCart } from './useCurrentCart';
+import { getInternationalizedStaticUrls } from 'helpers/getInternationalizedStaticUrls';
+import { useDomainConfig } from 'hooks/useDomainConfig';
+import { useRouter } from 'next/router';
+import { useEffect, useState } from 'react';
+
+export const useOrderPagesAccess = (page: 'transport-and-payment' | 'contact-information') => {
+    const router = useRouter();
+    const { cart, isFetching } = useCurrentCart();
+    const { url } = useDomainConfig();
+    const [canContentBeDisplayed, setCanContentBeDisplayed] = useState(false);
+    const [cartUrl, transportAndPaymentUrl] = getInternationalizedStaticUrls(
+        ['/cart', '/order/transport-and-payment'],
+        url,
+    );
+
+    const handleOrderPagesAccess = () => {
+        if (cart === undefined || isFetching) {
+            return;
+        }
+
+        if (cart === null || !cart.items.length) {
+            router.replace(cartUrl);
+        } else if (page === 'contact-information' && (!cart.transport || !cart.payment)) {
+            router.replace(transportAndPaymentUrl);
+        } else {
+            setCanContentBeDisplayed(true);
+        }
+    };
+
+    useEffect(() => {
+        handleOrderPagesAccess();
+    }, [cart, isFetching]);
+
+    return canContentBeDisplayed;
```

Comment:
Hmm, if I understand the logic right in case cart is `undefined `it will display skeleton even if the is no fetching happening. I think we should implement 3 possible values return -  boolean and `undefined` and then distinguish proper state. If return `undefined` that would mean cart is still fetching, boolean would only return when fetching is resolved. Otherwise we are not able distinguish between fetching and without access state.

Reply:
alright, but the skeleton would still be shown in case of a falsy value? I guess I have no problem with that, if it helps anything.

Reply:
Nope, skeleton would be shown only in case there is some fetching happening. This is simply missing handling for this case of undefined value. As user I don't want to see infinite skeleton in case there is failed action.

Reply:
Could you describe this more closely? I still do not understand what should happen in case of undefined. Not display anything? Or?

Reply:
We agreed on a call that this is okay